lib/apartment/adapters/abstract_adapter.rb

```diff
@@ -224,6 +231,34 @@ def rescuable_exceptions
       def rescue_from
         []
       end
+
+      def extended_cloned_config(tenant)
+        @config.clone.with_indifferent_access.tap do |config|
+          config.merge!(Apartment.db_config_for(tenant))
```

Comment:
Actually, config is taken from the current connection so I was wrong. Then, even if its verbose, it would be better to have people redefine everything, as suggested by @mikecmpbll , I'm going to change the code accordingly